Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPMediaSink.cpp

```diff
@@ -0,0 +1,211 @@
+    {
+        RETURNIFSHUTDOWN;
+        if (!pcStreamSinkCount)
+        {
+            return E_POINTER;
+        }
+        else
+        {
+            *pcStreamSinkCount = (DWORD)m_spStreamSinks.size();
+            return S_OK;
+        }
+    }
+
+    STDMETHODIMP GetStreamSinkByIndex(
+        /* [in] */ DWORD dwIndex,
+        /* [out] */ __RPC__deref_out_opt IMFStreamSink** ppStreamSink) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        if (!ppStreamSink) return E_POINTER;
+        if (dwIndex > 0)
+        {
+            return MF_E_INVALIDINDEX;
+        }
+        m_spStreamSinks[dwIndex].as<IMFStreamSink>().copy_to(ppStreamSink);
+        return S_OK;
+    }
+
+    STDMETHODIMP GetStreamSinkById(
+        /* [in] */ DWORD dwStreamSinkIdentifier,
+        /* [out] */ __RPC__deref_out_opt IMFStreamSink** ppStreamSink) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        if (dwStreamSinkIdentifier > m_spStreamSinks.size())
+        {
+            return MF_E_INVALIDSTREAMNUMBER;
+        }
+        else
+        {
+            m_spStreamSinks[dwStreamSinkIdentifier].as<IMFStreamSink>().copy_to(ppStreamSink);
+            return S_OK;
+        }
+    }
+
+    STDMETHODIMP SetPresentationClock(
+        /* [in] */ __RPC__in_opt IMFPresentationClock* pPresentationClock) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        if (m_spClock)
+        {
+            m_spClock->RemoveClockStateSink(this);
+            m_spClock = nullptr;
+        }
+        m_spClock.copy_from(pPresentationClock);
+        return m_spClock->AddClockStateSink(this);
+
+    }
+
+    STDMETHODIMP GetPresentationClock(
+        /* [out] */ __RPC__deref_out_opt IMFPresentationClock** ppPresentationClock) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        if (!ppPresentationClock) return E_POINTER;
+        m_spClock.copy_to(ppPresentationClock);
+        return S_OK;
+    }
+
+    STDMETHODIMP Shutdown(void) noexcept
+    {
+        RETURNIFSHUTDOWN;
+        HRESULT hr = S_OK;
+        m_bIsShutdown = true;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Shutdown();
+            hr = FAILED(hr1)? hr1: hr;
+        }
+        return hr;
+    }
+
+    //IMediaExtension
+    STDMETHODIMP SetProperties(
+        winrt::Windows::Foundation::Collections::IPropertySet const& configuration) noexcept
+    {
+        return S_OK;
+    }
+
+    // IMFClockStateSink methods
+    STDMETHODIMP OnClockStart(MFTIME hnsSystemTime, LONGLONG llClockStartOffset) noexcept
+    {
+        HRESULT hr = S_OK;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Start(hnsSystemTime, llClockStartOffset);
+            hr = FAILED(hr1) ? hr1 : hr;
+        }
+        return hr;
+    }
+    STDMETHODIMP OnClockStop(MFTIME hnsSystemTime) noexcept
+    {
+        HRESULT hr = S_OK;
+        for (auto strm : m_spStreamSinks)
+        {
+            auto hr1 = strm->Stop(hnsSystemTime);
+            hr = FAILED(hr1) ? hr1 : hr;

```

Comment:
>hr1 [](start = 24, length = 3)

should this be if succeeded(hr) so we don't overwrite the first error

Reply:
this logic remembers the last error that occurred and returns it.  This logic overrides the the first error only if there is a second error. In this logic, the success never overrides the failure. 
Anyway, this code path is kind of a  no-op , it doesn't return failure as per present code. 

---
In reply to: [491166891](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r491166891) [](ancestors = 491166891)